server/handlers/isomorphic-handler.js

```diff
@@ -384,15 +386,51 @@ exports.handleIsomorphicRoute = function handleIsomorphicRoute(
         return next();
       }
 
-      return new Promise(resolve => resolve(writeResponse(result)))
-        .catch(e => {
-          logError(e);
-          res.status(500);
-          res.send(e.message);
-        })
-        .finally(() => res.end());
+      const isAmpSupported = _.get(result, [
+        "data",
+        "story",
+        "is-amp-supported"
+      ]);
+      const ampStoryPagesConfig =
+        _.get(config, ["theme-attributes", "amp-story-pages"], false) ||
+        ampStoryPages;
+      if (
+        result.pageType === "story-page" &&
+        ampStoryPagesConfig &&
+        isAmpSupported
```

Comment:
instead of having this hardcoded here, the better thing would be something like the route having `renderLayout: false`. This way it's much cleaner and you don't have all this complex logic in the library